msu/calendar.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import print_function
+import pickle
+import os.path
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+from flask import current_app
+
+SCOPES = ['https://www.googleapis.com/auth/calendar']
+
+def check_for_token():
+    creds = None
+    if os.path.exists('token.pickle'):
+        with open('token.pickle', 'rb') as token:
+            creds = pickle.load(token)
+    return creds
+
+def build_service():
+    creds = check_for_token()
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                'credentials.json', SCOPES)
```

Comment:
You should rename the file to something more obvious (like `gcal_credentials.json`). We're going to have credentials for other services too, so it's good to disambiguate.

Reply:
Don't forget to update the `.gitignore` accordingly!